### ISSUE TITLE
Failed to activate jobs for type error on broker and Cannot apply index error

### ISSUE THREAD

**Issue by bhowmikdebarshi** [zeebe-io/zeebe#2952]
Hello

In one of our dev environment, we have the zeebe broker cluster with 3 nodes and partition count and replication count as 1.  The broker has been installed using 0.15.1 docker image on 3 separate ec2 machines. 
It was all working fine since last two weeks, but suddenly this morning I'm seeing following error in all the nodes - 

`07:02:18.531 [io.zeebe.gateway.impl.broker.BrokerRequestManager] [gateway-zb-actors-0] WARN  io.zeebe.gateway - Failed to activate jobs for type VIRUS_SCANNING from partition 0
io.zeebe.transport.RequestTimeoutException: Request timed out after PT15S
        at io.zeebe.transport.impl.sender.OutgoingRequest.timeout(OutgoingRequest.java:151) ~[zb-transport-0.15.1.jar:0.15.1]
        at io.zeebe.transport.impl.sender.Sender.onTimerExpiry(Sender.java:483) ~[zb-transport-0.15.1.jar:0.15.1]
        at org.agrona.DeadlineTimerWheel.poll(DeadlineTimerWheel.java:284) ~[agrona-0.9.32.jar:0.9.32]
        at io.zeebe.transport.impl.sender.Sender.processTimeouts(Sender.java:114) ~[zb-transport-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:90) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:53) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:187) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:154) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:135) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.run(ActorThread.java:112) [zb-util-0.15.1.jar:0.15.1]
07:02:20.271 [io.zeebe.gateway.impl.broker.BrokerRequestManager] [gateway-zb-actors-0] WARN  io.zeebe.gateway - Failed to activate jobs for type notificationProcessor from partition 0
io.zeebe.transport.RequestTimeoutException: Request timed out after PT15S
        at io.zeebe.transport.impl.sender.OutgoingRequest.timeout(OutgoingRequest.java:151) ~[zb-transport-0.15.1.jar:0.15.1]
        at io.zeebe.transport.impl.sender.Sender.onTimerExpiry(Sender.java:483) ~[zb-transport-0.15.1.jar:0.15.1]
        at org.agrona.DeadlineTimerWheel.poll(DeadlineTimerWheel.java:284) ~[agrona-0.9.32.jar:0.9.32]
        at io.zeebe.transport.impl.sender.Sender.processTimeouts(Sender.java:114) ~[zb-transport-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:90) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:53) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:187) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:154) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:135) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.run(ActorThread.java:112) [zb-util-0.15.1.jar:0.15.1]
07:02:20.502 [io.zeebe.gateway.impl.broker.BrokerRequestManager] [gateway-zb-actors-0] WARN  io.zeebe.gateway - Failed to activate jobs for type ocrProcessor from partition 0
io.zeebe.transport.RequestTimeoutException: Request timed out after PT15S
        at io.zeebe.transport.impl.sender.OutgoingRequest.timeout(OutgoingRequest.java:151) ~[zb-transport-0.15.1.jar:0.15.1]
        at io.zeebe.transport.impl.sender.Sender.onTimerExpiry(Sender.java:483) ~[zb-transport-0.15.1.jar:0.15.1]
        at org.agrona.DeadlineTimerWheel.poll(DeadlineTimerWheel.java:284) ~[agrona-0.9.32.jar:0.9.32]
        at io.zeebe.transport.impl.sender.Sender.processTimeouts(Sender.java:114) ~[zb-transport-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:90) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:53) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:187) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:154) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:135) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.run(ActorThread.java:112) [zb-util-0.15.1.jar:0.15.1]
07:02:20.502 [io.zeebe.gateway.impl.broker.BrokerRequestManager] [gateway-zb-actors-0] WARN  io.zeebe.gateway - Failed to activate jobs for type inferProcessor from partition 0
io.zeebe.transport.RequestTimeoutException: Request timed out after PT15S
        at io.zeebe.transport.impl.sender.OutgoingRequest.timeout(OutgoingRequest.java:151) ~[zb-transport-0.15.1.jar:0.15.1]
        at io.zeebe.transport.impl.sender.Sender.onTimerExpiry(Sender.java:483) ~[zb-transport-0.15.1.jar:0.15.1]
        at org.agrona.DeadlineTimerWheel.poll(DeadlineTimerWheel.java:284) ~[agrona-0.9.32.jar:0.9.32]
        at io.zeebe.transport.impl.sender.Sender.processTimeouts(Sender.java:114) ~[zb-transport-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:90) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:53) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:187) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:154) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:135) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.run(ActorThread.java:112) [zb-util-0.15.1.jar:0.15.1]
07:02:22.141 [io.zeebe.gateway.impl.broker.BrokerRequestManager] [gateway-zb-actors-0] WARN  io.zeebe.gateway - Failed to activate jobs for type updateDocumentMetadata from partition 0
io.zeebe.transport.RequestTimeoutException: Request timed out after PT15S
        at io.zeebe.transport.impl.sender.OutgoingRequest.timeout(OutgoingRequest.java:151) ~[zb-transport-0.15.1.jar:0.15.1]
        at io.zeebe.transport.impl.sender.Sender.onTimerExpiry(Sender.java:483) ~[zb-transport-0.15.1.jar:0.15.1]
        at org.agrona.DeadlineTimerWheel.poll(DeadlineTimerWheel.java:284) ~[agrona-0.9.32.jar:0.9.32]
        at io.zeebe.transport.impl.sender.Sender.processTimeouts(Sender.java:114) ~[zb-transport-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:90) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:53) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:187) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:154) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:135) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.run(ActorThread.java:112) [zb-util-0.15.1.jar:0.15.1]
07:02:26.514 [io.zeebe.gateway.impl.broker.BrokerRequestManager] [gateway-zb-actors-0] ERROR io.zeebe.gateway - Error handling gRPC request
io.grpc.StatusRuntimeException: INTERNAL: Unexpected error occurred during the request processing
        at io.grpc.Status.asRuntimeException(Status.java:521) ~[grpc-core-1.18.0.jar:1.18.0]
        at io.zeebe.gateway.EndpointManager.convertThrowable(EndpointManager.java:258) ~[zeebe-gateway-0.15.1.jar:0.15.1]
        at io.zeebe.gateway.EndpointManager.lambda$sendRequest$2(EndpointManager.java:236) ~[zeebe-gateway-0.15.1.jar:0.15.1]
        at io.zeebe.gateway.impl.broker.BrokerRequestManager.lambda$sendRequest$3(BrokerRequestManager.java:118) ~[zeebe-gateway-0.15.1.jar:0.15.1]
        at io.zeebe.gateway.impl.broker.BrokerRequestManager.lambda$sendRequestInternal$6(BrokerRequestManager.java:193) ~[zeebe-gateway-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.future.FutureContinuationRunnable.run(FutureContinuationRunnable.java:40) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:90) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:53) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:187) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:154) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:135) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.run(ActorThread.java:112) [zb-util-0.15.1.jar:0.15.1]
Caused by: io.zeebe.transport.RequestTimeoutException: Request timed out after PT15S
        at io.zeebe.transport.impl.sender.OutgoingRequest.timeout(OutgoingRequest.java:151) ~[zb-transport-0.15.1.jar:0.15.1]
        at io.zeebe.transport.impl.sender.Sender.onTimerExpiry(Sender.java:483) ~[zb-transport-0.15.1.jar:0.15.1]
        at org.agrona.DeadlineTimerWheel.poll(DeadlineTimerWheel.java:284) ~[agrona-0.9.32.jar:0.9.32]
        at io.zeebe.transport.impl.sender.Sender.processTimeouts(Sender.java:114) ~[zb-transport-0.15.1.jar:0.15.1]
        ... 6 more
07:02:27.780 [io.zeebe.gateway.impl.broker.BrokerRequestManager] [gateway-zb-actors-0] WARN  io.zeebe.gateway - Failed to activate jobs for type fetchDocumentMetadata from partition 0
io.zeebe.transport.RequestTimeoutException: Request timed out after PT15S
        at io.zeebe.transport.impl.sender.OutgoingRequest.timeout(OutgoingRequest.java:151) ~[zb-transport-0.15.1.jar:0.15.1]
        at io.zeebe.transport.impl.sender.Sender.onTimerExpiry(Sender.java:483) ~[zb-transport-0.15.1.jar:0.15.1]
        at org.agrona.DeadlineTimerWheel.poll(DeadlineTimerWheel.java:284) ~[agrona-0.9.32.jar:0.9.32]
        at io.zeebe.transport.impl.sender.Sender.processTimeouts(Sender.java:114) ~[zb-transport-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:90) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:53) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:187) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:154) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:135) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.run(ActorThread.java:112) [zb-util-0.15.1.jar:0.15.1]
07:02:27.781 [io.zeebe.gateway.impl.broker.BrokerRequestManager] [gateway-zb-actors-0] WARN  io.zeebe.gateway - Failed to activate jobs for type createDocument from partition 0
io.zeebe.transport.RequestTimeoutException: Request timed out after PT15S
        at io.zeebe.transport.impl.sender.OutgoingRequest.timeout(OutgoingRequest.java:151) ~[zb-transport-0.15.1.jar:0.15.1]
        at io.zeebe.transport.impl.sender.Sender.onTimerExpiry(Sender.java:483) ~[zb-transport-0.15.1.jar:0.15.1]
        at org.agrona.DeadlineTimerWheel.poll(DeadlineTimerWheel.java:284) ~[agrona-0.9.32.jar:0.9.32]
        at io.zeebe.transport.impl.sender.Sender.processTimeouts(Sender.java:114) ~[zb-transport-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:90) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:53) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:187) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:154) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:135) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.run(ActorThread.java:112) [zb-util-0.15.1.jar:0.15.1]
07:02:27.781 [io.zeebe.gateway.impl.broker.BrokerRequestManager] [gateway-zb-actors-0] WARN  io.zeebe.gateway - Failed to activate jobs for type bulkUpload from partition 0
io.zeebe.transport.RequestTimeoutException: Request timed out after PT15S
        at io.zeebe.transport.impl.sender.OutgoingRequest.timeout(OutgoingRequest.java:151) ~[zb-transport-0.15.1.jar:0.15.1]
        at io.zeebe.transport.impl.sender.Sender.onTimerExpiry(Sender.java:483) ~[zb-transport-0.15.1.jar:0.15.1]
        at org.agrona.DeadlineTimerWheel.poll(DeadlineTimerWheel.java:284) ~[agrona-0.9.32.jar:0.9.32]
        at io.zeebe.transport.impl.sender.Sender.processTimeouts(Sender.java:114) ~[zb-transport-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:90) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:53) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:187) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:154) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:135) [zb-util-0.15.1.jar:0.15.1]
        at io.zeebe.util.sched.ActorThread.run(ActorThread.java:112) [zb-util-0.15.1.jar:0.15.1]`

Checked the disk spaces on all nodes and those look good. Only 3%-6% usage. 
Checked the Job handlers ( java clients) and those are up and running. 

I have seen this error previously and restarting all the nodes had resolved it, but would like to know what is the root cause for this error, or if I can get any direction where to look at, that would be a great help.

**Comment by npepinpe** [comment]
It looks like the command API is unresponsive and the requests from the gateway to the broker are timing out. From the log it's hard to say much more. 

Before investigating more, however, is it possible at all for you to upgrade to the latest version, 0.20.0? It is our [first production ready release](https://zeebe.io/blog/2019/07/announcing-zeebe-0-20-production-ready/), and contains many bug fixes.

**Comment by bhowmikdebarshi** [comment]
Thanks. 
Yes we are trying to upgrade to 0.20.0 in parallel and will see if this error occurs. 
But I'm still skeptical of having it fixed after upgrading. We have another dev environment with same zeebe version and same way of deployment, where we dint see this issue. But on this particular environment, it appeared twice in six weeks. 
So trying to think of the possible reasons of gateway and broker communication error while both of the components are in same machine.

**Comment by bhowmikdebarshi** [comment]
Hello, we have upgraded to 0.20.0 version. But still seeing this error. 

Here is the cluster status - 

`root@ip-10-35-230-148:/usr/local/zeebe/bin# zbctl status
Cluster size: 3
Partitions count: 3
Replication factor: 1
Brokers:
  Broker 0 - 10.35.231.206:26501
    Partition 1 : Leader
  Broker 1 - 10.35.230.148:26501
    Partition 2 : Leader
  Broker 2 - 10.35.234.200:26501
    Partition 3 : Leader
`
- which looks good. 

But I'm seeing following error on Broker 1 (node 1) log - 

`06:53:49.413 [io.zeebe.gateway.impl.broker.BrokerRequestManager] [10.35.230.148:26501-zb-actors-0] WARN io.zeebe.gateway - Failed to activate jobs for type ocrProcessor from partition 2
io.zeebe.transport.RequestTimeoutException: Request timed out after PT15S
at io.zeebe.transport.impl.sender.OutgoingRequest.timeout(OutgoingRequest.java:143) ~[zeebe-transport-0.20.0.jar:0.20.0]
at io.zeebe.transport.impl.sender.Sender.onTimerExpiry(Sender.java:475) ~[zeebe-transport-0.20.0.jar:0.20.0]
at org.agrona.DeadlineTimerWheel.poll(DeadlineTimerWheel.java:284) ~[agrona-1.0.3.jar:1.0.3]
at io.zeebe.transport.impl.sender.Sender.processTimeouts(Sender.java:106) ~[zeebe-transport-0.20.0.jar:0.20.0]
at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:80) [zeebe-util-0.20.0.jar:0.20.0]
at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:43) [zeebe-util-0.20.0.jar:0.20.0]
at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:169) [zeebe-util-0.20.0.jar:0.20.0]
at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:134) [zeebe-util-0.20.0.jar:0.20.0]
at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:118) [zeebe-util-0.20.0.jar:0.20.0]
at io.zeebe.util.sched.ActorThread.run(ActorThread.java:96) [zeebe-util-0.20.0.jar:0.20.0]` 

Also seeing following error on the same node - 

`06:53:49.475 [] [raft-server-raft-atomix-partition-2] ERROR io.atomix.protocols.raft.impl.RaftServiceManager - RaftServer{raft-atomix-partition-2} - Cannot apply index 46553713
06:53:49.475 [] [raft-server-raft-atomix-partition-2] ERROR io.atomix.protocols.raft.impl.RaftServiceManager - RaftServer{raft-atomix-partition-2} - Cannot apply index 46553714`

**Comment by bhowmikdebarshi** [comment]
Some more log from 29th log file regarding index error - 

`01:12:37.065 [] [raft-server-raft-atomix-partition-2] ERROR io.atomix.protocols.raft.impl.RaftServiceManager - RaftServer{raft-atomix-partition-2} - Cannot apply index 46424369
01:12:37.066 [] [raft-server-raft-atomix-partition-2] WARN  io.atomix.protocols.raft.roles.LeaderRole - RaftServer{raft-atomix-partition-2}{role=LEADER} - An unexpected error occurred: {}
java.lang.IndexOutOfBoundsException: Cannot apply index 46424369
        at io.atomix.protocols.raft.impl.RaftServiceManager.applyIndex(RaftServiceManager.java:398) ~[atomix-raft-3.2.0-alpha4.jar:?]
        at io.atomix.protocols.raft.impl.RaftServiceManager.lambda$enqueueIndex$9(RaftServiceManager.java:360) ~[atomix-raft-3.2.0-alpha4.jar:?]
        at io.atomix.utils.concurrent.SingleThreadContext$1.lambda$execute$0(SingleThreadContext.java:53) ~[atomix-utils-3.2.0-alpha4.jar:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_212]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_212]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_212]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_212]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_212]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_212]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212]
01:12:37.221 [] [raft-server-raft-atomix-partition-2] ERROR io.atomix.protocols.raft.impl.RaftServiceManager - RaftServer{raft-atomix-partition-2} - Cannot apply index 46424370
01:12:37.221 [] [raft-server-raft-atomix-partition-2] WARN  io.atomix.protocols.raft.roles.LeaderRole - RaftServer{raft-atomix-partition-2}{role=LEADER} - An unexpected error occurred: {}
java.lang.IndexOutOfBoundsException: Cannot apply index 46424370
        at io.atomix.protocols.raft.impl.RaftServiceManager.applyIndex(RaftServiceManager.java:398) ~[atomix-raft-3.2.0-alpha4.jar:?]
        at io.atomix.protocols.raft.impl.RaftServiceManager.lambda$enqueueIndex$9(RaftServiceManager.java:360) ~[atomix-raft-3.2.0-alpha4.jar:?]
        at io.atomix.utils.concurrent.SingleThreadContext$1.lambda$execute$0(SingleThreadContext.java:53) ~[atomix-utils-3.2.0-alpha4.jar:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_212]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_212]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_212]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_212]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_212]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_212]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212]
01:12:37.353 [] [raft-server-raft-atomix-partition-2] ERROR io.atomix.protocols.raft.impl.RaftServiceManager - RaftServer{raft-atomix-partition-2} - Cannot apply index 46424371
01:12:37.354 [] [raft-server-raft-atomix-partition-2] WARN  io.atomix.protocols.raft.roles.LeaderRole - RaftServer{raft-atomix-partition-2}{role=LEADER} - An unexpected error occurred: {}
java.lang.IndexOutOfBoundsException: Cannot apply index 46424371
        at io.atomix.protocols.raft.impl.RaftServiceManager.applyIndex(RaftServiceManager.java:398) ~[atomix-raft-3.2.0-alpha4.jar:?]
        at io.atomix.protocols.raft.impl.RaftServiceManager.lambda$enqueueIndex$9(RaftServiceManager.java:360) ~[atomix-raft-3.2.0-alpha4.jar:?]
        at io.atomix.utils.concurrent.SingleThreadContext$1.lambda$execute$0(SingleThreadContext.java:53) ~[atomix-utils-3.2.0-alpha4.jar:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_212]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_212]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_212]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_212]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_212]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_212]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212]
01:12:37.359 [] [raft-server-raft-atomix-partition-2] ERROR io.atomix.protocols.raft.impl.RaftServiceManager - RaftServer{raft-atomix-partition-2} - Cannot apply index 46424372
01:12:37.359 [] [raft-server-raft-atomix-partition-2] WARN  io.atomix.protocols.raft.roles.LeaderRole - RaftServer{raft-atomix-partition-2}{role=LEADER} - An unexpected error occurred: {}
java.lang.IndexOutOfBoundsException: Cannot apply index 46424372
        at io.atomix.protocols.raft.impl.RaftServiceManager.applyIndex(RaftServiceManager.java:398) ~[atomix-raft-3.2.0-alpha4.jar:?]
        at io.atomix.protocols.raft.impl.RaftServiceManager.lambda$enqueueIndex$9(RaftServiceManager.java:360) ~[atomix-raft-3.2.0-alpha4.jar:?]
        at io.atomix.utils.concurrent.SingleThreadContext$1.lambda$execute$0(SingleThreadContext.java:53) ~[atomix-utils-3.2.0-alpha4.jar:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_212]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_212]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_212]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_212]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_212]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_212]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212]
01:12:37.662 [] [raft-server-raft-atomix-partition-2] ERROR io.atomix.protocols.raft.impl.RaftServiceManager - RaftServer{raft-atomix-partition-2} - Cannot apply index 46424373
01:12:37.884 [] [raft-server-raft-atomix-partition-2] ERROR io.atomix.protocols.raft.impl.RaftServiceManager - RaftServer{raft-atomix-partition-2} - Cannot apply index 46424374
01:12:37.887 [] [raft-server-raft-atomix-partition-2] ERROR io.atomix.protocols.raft.impl.RaftServiceManager - RaftServer{raft-atomix-partition-2} - Cannot apply index 46424375
01:12:37.891 [] [raft-server-raft-atomix-partition-2] ERROR io.atomix.protocols.raft.impl.RaftServiceManager - RaftServer{raft-atomix-partition-2} - Cannot apply index 46424376
01:12:37.892 [] [raft-server-raft-atomix-partition-2] ERROR io.atomix.protocols.raft.impl.RaftServiceManager - RaftServer{raft-atomix-partition-2} - Cannot apply index 46424377
01:12:37.893 [] [raft-server-raft-atomix-partition-2] ERROR io.atomix.protocols.raft.impl.RaftServiceManager - RaftServer{raft-atomix-partition-2} - Cannot apply index 46424378
`

**Comment by bhowmikdebarshi** [comment]
`ERROR io.atomix.protocols.raft.impl.RaftServiceManager - RaftServer{raft-atomix-partition-2} - Cannot apply index`

This error is coming too frequently. Restarting the node is as of now only solution but it again comes within few hours. Not sure what might be the reason of this error. 

Any help would be appreciated. 

Thanks

**Comment by bhowmikdebarshi** [comment]
Closing this one as raised a separate issue #3073 with the latest error
